cmd/endpoint-ellipses.go

```diff
@@ -129,6 +112,44 @@ func getSetIndexes(args []string, totalSizes []uint64, customSetDriveCount uint6
 			return nil, config.ErrInvalidErasureSetSize(nil).Msg(msg)
 		}
 		setSize = customSetDriveCount
+		globalCustomErasureDriveCount = true
+	} else {
+		var newSetCounts = make(map[uint64]struct{})
+		for _, ss := range setCounts {
+			var valid bool
+			for _, argPattern := range argPatterns {
+				for _, p := range argPattern {
+					if uint64(len(p.Seq)) > ss {
+						valid = uint64(len(p.Seq))%ss == 0
+					} else {
+						valid = ss%uint64(len(p.Seq)) == 0
+					}
+				}
+			}
+			if _, ok := newSetCounts[ss]; !ok && (valid || argPatterns == nil) {
+				newSetCounts[ss] = struct{}{}
+			}
+		}
+
+		setCounts = []uint64{}
+		for setCount := range newSetCounts {
```

Comment:
I can't say I understand everything, but it seems like a potential problem that the range is in random order.

Reply:
Generally has no bearing on the final result, because order is automatically obtained due to the subsequent divisibility factors obtained. But it is good to sort them..